src/redux/ui/stats/actions.ts

```diff
@@ -0,0 +1,47 @@
+import { Dispatch } from 'react';
+import { getBlockchain } from '../../../core/blockchain/blockchain-factory';
+import { getChainId } from '../../preferences/selectors';
+import { IReduxState } from '../../state';
+import { IAccountState, ITokenState } from '../../wallets/state';
+import { captureException as SentryCaptureException } from '@sentry/react-native';
+import { getAccountStatsTimestamp } from './selectors';
+import { FETCH_ACCOUNT_STATS_SECONDS } from '../../../core/constants/app';
+import moment from 'moment';
+
+export const ADD_ACCOUNT_STATS = 'ADD_ACCOUNT_STATS';
+
+export const fetchAccountDelegateStats = (account: IAccountState, token: ITokenState) => async (
+    dispatch: Dispatch<any>,
+    getState: () => IReduxState
+) => {
+    const state = getState();
+    const blockchain = account.blockchain;
+    const chainId = getChainId(state, blockchain).toString();
+
+    try {
+        const timestamp = getAccountStatsTimestamp(state, blockchain, chainId, account.address);
+
+        if (
+            !timestamp ||
+            moment().isAfter(moment(timestamp).add(FETCH_ACCOUNT_STATS_SECONDS, 'seconds'))
```

Comment:
you like moment a lot, but in some cases it's quite easy to use simple verifications. For instance in this case you could do this:
```
timestamp < Date.now() - FETCH_ACCOUNT_STATS_SECONDS * 1000
```

Reply:
you don't have to do the changes, i'm ok with your solution, it's just a hint for future :)